laptop/setup.sh

```diff
@@ -45,38 +45,61 @@ update_apt() {
 	apt-get update
 }
 
-install_k3d() {
-	local k3d_Version=$1
-
-	wget -q -O - https://raw.githubusercontent.com/k3d-io/k3d/main/install.sh | TAG="$k3d_version" bash
-}
-
-start_k3d() {
-    echo "Creating the cluster..."
-    k3d cluster create --network host --no-lb --k3s-arg "--disable=traefik,servicelb" --k3s-arg "--kube-apiserver-arg=feature-gates=MixedProtocolLBService=true" --host-pid-mode
-
-    echo "Waiting for the cluster to be fully operational..."
-    sleep 10
-
-    echo "Configuring kubeconfig..."
-    mkdir -p ~/.kube/
-
-    k3d kubeconfig get -a > ~/.kube/config || echo "Failed to get kubeconfig"
-
-    echo "Checking nodes..."
-    until kubectl wait --for=condition=Ready nodes --all --timeout=600s; do
-        echo "Waiting for nodes to be ready..."
-        sleep 5
-    done
+start_k3s() {
+	local k3s_version=$1
+	echo "Creating the cluster..."
+
+	docker run -d --privileged --name ctrlplane-laptop \
+		-e K3S_KUBECONFIG_OUTPUT=/output/kubeconfig.yaml \
+		-e K3S_KUBECONFIG_MODE=666 \
+		-v "$(pwd):/output" \
+		-v k3s-server:/var/lib/rancher/k3s \
+		--tmpfs=/run --tmpfs=/var/run \
+		--network=host \
+		rancher/k3s:"$k3s_version" server \
+		--disable=traefik,servicelb \
+		--tls-san=ctrlplane-laptop \
+		--node-label="colony.konstruct.io/node-type=laptop"
+
+	echo "Waiting for the cluster to be fully operational..."
+	sleep 30
+
+	echo "Configuring kubeconfig..."
+	mkdir -p ~/.kube/
+
+	cp ./kubeconfig.yaml ~/.kube/config || echo "Failed to get kubeconfig"
+	sed -i 's/127.0.0.1/localhost/g' ~/.kube/config
```

Comment:
Curious to know what your setup looks like that 127.0.0.1 (aka "localhost") didn't work.

Reply:
I'll test this part again, you have a point

Reply:
removed